twill-core/src/test/java/org/apache/twill/internal/utils/ApplicationBundlerTest.java

```diff
@@ -68,6 +71,56 @@ public void testFindDependencies() throws IOException, ClassNotFoundException {
     Assert.assertNotSame(classLoader, clz.getClassLoader());
   }
 
+  @Test
+  public void testSameJar() throws IOException, ClassNotFoundException {
+    File dir1 = tmpDir.newFolder();
+    File dir2 = tmpDir.newFolder();
+    File j1 = new File(dir1, "samename.jar");
+    File j2 = new File(dir2, "samename.jar");
+
+    createJar(Class1.class, j1);
+    createJar(Class2.class, j2);
+    
+    ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
+    Location location;
+
+    try {
+      URL[] urls = new URL[] { j1.toURI().toURL(), j2.toURI().toURL() };
+      Thread.currentThread().setContextClassLoader(new URLClassLoader(urls, null));
+
+      // create bundle
+      location = new LocalLocationFactory(tmpDir.newFolder()).create("test.jar");
+      ApplicationBundler bundler = new ApplicationBundler(ImmutableList.<String> of());
+      bundler.createBundle(location, Class1.class, Class2.class);
+
+    } finally {
+      Thread.currentThread().setContextClassLoader(currentClassLoader);
+    }
+
+    File targetDir = tmpDir.newFolder();
+    unjar(new File(location.toURI()), targetDir);
+
+    // should be able to load both classes
+    ClassLoader classLoader = createClassLoader(targetDir);
+    Class<?> c1 = classLoader.loadClass(Class1.class.getName());
+    Class<?> c2 = classLoader.loadClass(Class2.class.getName());
+
+    // make sure we are loading them from the correct class loader (not defaulting back to some classloader
+    // from the test
+    Assert.assertSame(classLoader, c1.getClassLoader());
+    Assert.assertSame(classLoader, c2.getClassLoader());
+  }
+
+  private void createJar(Class<?> clazz, File jarFile) throws IOException {
+    try (JarOutputStream jos = new JarOutputStream(new FileOutputStream(jarFile))) {
+      String pathname = clazz.getName().replace(".", "/") + ".class";
+      JarEntry entry = new JarEntry(pathname);
+      jos.putNextEntry(entry);
+      IOUtils.copy(clazz.getClassLoader().getResourceAsStream(pathname), jos);
```

Comment:
Sorry I missed this in last review. Usually we use guava in twill instead of apache common. For this, can you use `Resources.copy(clazz.getClassLoader().getResource(pathname), jos);` instead? Thanks.